plutus-contract/test/Spec/TxConstraints/MustPayToPubKeyAddress.hs

```diff
@@ -0,0 +1,265 @@
+import Plutus.Trace qualified as Trace
+import Plutus.V1.Ledger.Api (CurrencySymbol (CurrencySymbol), Datum (Datum), Redeemer (Redeemer),
+                             ToData (toBuiltinData), UnsafeFromData (unsafeFromBuiltinData))
+import Plutus.V1.Ledger.Value qualified as Value
+import PlutusTx qualified
+import PlutusTx.Prelude qualified as P
+import Wallet.Emulator.Wallet (walletToMockWallet')
+
+tests :: TestTree
+tests =
+    testGroup "MustPayToPubKeyAddress"
+        [ successfulUseOfMustPayToPubKeyWithExactTokenValue
+        , successfulUseOfMustPayToPubKeyExpectingALowerAdaValue
+        , successfulUseOfMustPayToPubKeyAddress
+        , successfulUseOfMustPayWithDatumToPubKey
+        , successfulUseOfMustPayWithDatumToPubKeyAddress
+        , phase2FailureWhenUsingUnexpectedPaymentPubKeyHash
+        --, phase2FailureWhenUsingUnexpectedStakePubKeyHash -- FAILING
+        , phase2FailureWhenUsingUnexpectedDatum
+        , phase2FailureWhenUsingUnexpectedValue
+         ]
+
+someDatum :: Datum
+someDatum = Datum $ PlutusTx.dataToBuiltinData $ PlutusTx.toData ("datum" :: P.BuiltinByteString)
+
+otherDatum :: Datum
+otherDatum = Datum $ PlutusTx.dataToBuiltinData $ PlutusTx.toData ("other datum" :: P.BuiltinByteString)
+
+adaValue :: Value.Value
+adaValue = Ada.lovelaceValueOf 5_000_000
+
+tknValue :: Value.Value
+tknValue = Value.singleton mustPayToPubKeyAddressPolicyCurrencySymbol "mint-me" 1
+
+otherTknValue :: Value.Value
+otherTknValue = Value.singleton mustPayToPubKeyAddressPolicyCurrencySymbol "dont-mint-me" 1
+
+w1PaymentPubKeyHash :: Ledger.PaymentPubKeyHash
+w1PaymentPubKeyHash = mockWalletPaymentPubKeyHash w1
+
+w2PaymentPubKeyHash :: Ledger.PaymentPubKeyHash
+w2PaymentPubKeyHash = mockWalletPaymentPubKeyHash w2
+
+w1StakePubKeyHash :: Ledger.StakePubKeyHash
+w1StakePubKeyHash = fromJust $ stakePubKeyHash $ walletToMockWallet' w1
+
+w2StakePubKeyHash :: Ledger.StakePubKeyHash
+w2StakePubKeyHash = fromJust $ stakePubKeyHash $ walletToMockWallet' w2
+
+asRedeemer :: PlutusTx.ToData a => a -> Redeemer
+asRedeemer a = Redeemer $ PlutusTx.dataToBuiltinData $ PlutusTx.toData a
+
+trace :: Contract () Empty ContractError () -> Trace.EmulatorTrace ()
+trace contract = do
+    void $ Trace.activateContractWallet w1 contract
+    void $ Trace.waitNSlots 1
+
+-- | Valid scenario using offchain and onchain constraint mustPayToPubKey with exact token value
+successfulUseOfMustPayToPubKeyWithExactTokenValue :: TestTree
+successfulUseOfMustPayToPubKeyWithExactTokenValue =
+    let options = defaultCheckOptions & changeInitialWalletValue w1 (<> tknValue)
+        onChainConstraint = asRedeemer $ MustPayToPubKey w1PaymentPubKeyHash tknValue
+        contract = do
+            let lookups1 = Constraints.plutusV1MintingPolicy mustPayToPubKeyAddressPolicy
+                tx1 = Constraints.mustPayToPubKey w1PaymentPubKeyHash adaValue
+                   <> Constraints.mustMintValueWithRedeemer onChainConstraint tknValue
```

Comment:
Why work with a minting script instead of a validator script?

Reply:
No strong reasons but it does makes the contracts shorter